azurerm/resource_arm_automation_schedule.go

```diff
@@ -282,3 +364,90 @@ func resourceArmAutomationScheduleDelete(d *schema.ResourceData, meta interface{
 
 	return nil
 }
+
+func expandArmAutomationScheduleAdvanced(v []interface{}) (*automation.AdvancedSchedule, error) {
+
+	// Get the one and only advance schedule configuration
+	advancedSchedule := v[0].(map[string]interface{})
+	weekDays := advancedSchedule["week_days"].(*schema.Set).List()
+	monthDays := advancedSchedule["month_days"].(*schema.Set).List()
+	monthlyOccurrences := advancedSchedule["monthly_occurrence"].([]interface{})
+
+	expandedAdvancedSchedule := automation.AdvancedSchedule{}
+
+	if len(weekDays) > 0 {
+		expandedWeekDays := make([]string, len(weekDays))
+		for i := range weekDays {
+			expandedWeekDays[i] = weekDays[i].(string)
+		}
+		expandedAdvancedSchedule.WeekDays = &expandedWeekDays
+	}
+
+	if len(monthDays) > 0 {
+		expandedMonthDays := make([]int32, len(monthDays))
+		for i := range monthDays {
+			expandedMonthDays[i] = int32(monthDays[i].(int))
+		}
+		expandedAdvancedSchedule.MonthDays = &expandedMonthDays
+	}
+
+	if len(monthlyOccurrences) > 0 {
+		expandedMonthlyOccurrences := make([]automation.AdvancedScheduleMonthlyOccurrence, len(monthlyOccurrences))
+		for i := range monthlyOccurrences {
+			m := monthlyOccurrences[i].(map[string]interface{})
+			occurrence := int32(m["occurrence"].(int))
+
+			expandedMonthlyOccurrences[i] = automation.AdvancedScheduleMonthlyOccurrence{
+				Occurrence: &occurrence,
+				Day:        automation.ScheduleDay(m["day"].(string)),
+			}
+		}
+		expandedAdvancedSchedule.MonthlyOccurrences = &expandedMonthlyOccurrences
+	}
+
+	return &expandedAdvancedSchedule, nil
+}
+
+func flattenArmAutomationScheduleAdvanced(v *automation.AdvancedSchedule) []interface{} {
+	result := make(map[string]interface{})
+
+	if v.WeekDays != nil {
+		flattenedWeekDays := schema.NewSet(set.HashStringIgnoreCase, []interface{}{})
+		for i := range *v.WeekDays {
+			flattenedWeekDays.Add((*v.WeekDays)[i])
+		}
+		result["week_days"] = flattenedWeekDays
+	}
```

Comment:
can we update this to be:

```
flattenedWeekDays := schema.NewSet(set.HashStringIgnoreCase, []interface{}{})
if v.WeekDays != nil {
  for _, v := range *v.WeekDays {
    flattenedWeekDays.Add(v)
  }
}
result["week_days"] = flattenedWeekDays
```

this allows this to be set even if nil's returned from the API - which means diff's will be shown